### ISSUE TITLE
Document why `skip-prune` is not defailt

### ISSUE THREAD

**Issue by jamietanna** [deepmap/oapi-codegen#1442]
Via:

>              Probably not for this PR, but it would make sense to not prune code that is used in dependent APIs by default, instead of having to set this explicitly.

_Originally posted by @markuswustenberg in https://github.com/deepmap/oapi-codegen/pull/1389#discussion_r1466286582_

**Comment by jamietanna** [comment]
This is called out in https://github.com/deepmap/oapi-codegen?tab=readme-ov-file#key-design-decisions - do we want any more than this?

**Comment by markuswustenberg** [comment]
The issue is about not pruning code that is used in other, dependent APIs, where those APIs are known when generating, I think? In which case it would make sense to not prune them by default, because they're used in the dependent APIs, and it otherwise leads to errors in the generated code.

TLDR: if an option is not optional but mandatory for the generated code to work, why is it an option?

**Comment by jmgilman** [comment]
Just adding some feedback here. My use-case is turning OpenAPI specs generated by CUE into Go structs. For example:

```
{
    "openapi": "3.0.0",
    "info": {
        "title": "Generated by cue.",
        "version": "no version"
    },
    "paths": {},
    "components": {
        "schemas": {
            "Schema": {
                "type": "object",
                "required": [
                    "registry"
                ],
                "properties": {
                    "registry": {
                        "type": "string"
                    }
                }
            }
        }
    }
}
```

The default behavior, in this case, results in an empty file being generated. Given the length of the README, it took me a while before I realized I was missing the `skip-prune` setting. Regardless of whether it's defaulted true or not, maybe an explicit example like above with a working configuration in the README would go a long ways towards saving someone this much effort in the future.
